### ISSUE TITLE
Sync issue

### ISSUE THREAD

**Issue by lemieuxl** [memen45/SubMusic#45]
I've got a sync issue which looks a bit like #30. I'm using SubMusic v0.1.24-yankee on a Venu watch with Ampache 5.3.1.

Initial sync always works fine and songs get transferred to the watch. Then, when auto-sync starts (when watch is plugged in), sync stays at 0%.

Apache log shows those two lines again and again (like in #30).

```text
192.168.X.X - - [04/May/2022:16:07:10 -0400] "GET /server/json.server.php?date=1651160598&id=13&client=SubMusic-v0.1.24-yankee&action=record_play&auth=XXXX HTTP/1.1" 200 103 "-" "Garmin Venu 2/8.11"
192.168.X.X - - [04/May/2022:16:07:10 -0400] "GET /server/json.server.php?auth=XXXX&timestamp=1651694829&action=handshake&user=XXXX&version=440000 HTTP/1.1" 200 512 "-" "Garmin Venu 2/8.11"
```

Looking at ampache log, I see the following (over and over again).

```text
2022-05-04T20:11:27+00:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-05-04T20:11:27+00:00 [] (Ampache\Module\Api\Authentication\Gatekeeper) -> API session [XXXX]
2022-05-04T20:11:27+00:00 [] (Ampache\Module\Api\ApiHandler) -> API function [Ampache\Module\Api\Method\Api4\Handshake4Method]
2022-05-04T20:11:27+00:00 [] (Ampache\Module\Api\Method\Api4\Handshake4Method) -> Login4 Attempt, IP: 192.168.X.X Time: 1651695083 User: USER (3)
2022-05-04T20:11:27+00:00 [] (Ampache\Module\System\Session) -> Unable to read value from key  no data found
2022-05-04T20:11:27+00:00 [] (Ampache\Module\System\Session) -> Session created: YYYY
2022-05-04T20:11:27+00:00 [] (Ampache\Module\Api\Method\Api4\Handshake4Method) -> Login Success, passphrase matched
2022-05-04T20:11:27+00:00 [] (Ampache\Module\System\Session) -> XXXX has been extended to Wed, 04 May 2022 21:11:27 +0000 extension length 3600
2022-05-04T20:11:27+00:00 [] (Ampache\Module\System\Dba) -> Database connection...
2022-05-04T20:11:27+00:00 [] (Ampache\Module\Api\Authentication\Gatekeeper) -> API session [YYYY]
2022-05-04T20:11:27+00:00 [USER] (Ampache\Module\Api\ApiHandler) -> API function [Ampache\Module\Api\Method\Api4\RecordPlay4Method]
2022-05-04T20:11:27+00:00 [ampache] (log.lib) -> [Runtime Error] Undefined array key "user" in file /var/www/ampache-5.3.1/src/Module/Api/Api4.php(218)
2022-05-04T20:11:27+00:00 [ampache] (log.lib) -> [Runtime Error] Undefined array key "user" in file /var/www/ampache-5.3.1/src/Module/Api/Api4.php(218)
2022-05-04T20:11:27+00:00 [USER] (Ampache\Module\Api\Api4) -> 'user' required on record_play function call.
2022-05-04T20:11:27+00:00 [USER] (Ampache\Module\System\Session) -> YYYY has been extended to Wed, 04 May 2022 21:11:27 +0000 extension length 3600
```

**Comment by memen45** [comment]
Thanks for reporting your issue! I opened an issue at Ampache/ampache#3252 as based on the PHP code, it seems the same error as before was reintroduced in the Ampache codebase. 

I am not sure why SubMusic keeps retrying. Can you see in the logs what exactly the response is for the record_play request and the handshake request? Of course obfuscate the actual user info, I'm interested in the session expiry and time values.
The session should normally be extended only once, and after that, no handshake requests should be made. Do you see many handshake requests always, or only when this record_play issue shows?

**Comment by lemieuxl** [comment]
I honestly don't know how to log the response (header nor body) from httpd... I tried _mod_dumpio_ and _mod_security_ without success.

This is the beginning of the log once the sync starts (and hangs). It goes `handshake` then `record_play` multiple times per second.

```text
192.168.X.X - - [05/May/2022:09:01:51 -0400] "GET /server/json.server.php?auth=XXXX&timestamp=1651755707&action=handshake&user=USER&version=440000 HTTP/1.1" 200 512 "-" "Garmin Venu 2/8.11"
192.168.X.X - - [05/May/2022:09:01:51 -0400] "GET /server/json.server.php?client=SubMusic-v0.1.24-yankee&id=13&date=1651160598&action=record_play&auth=AAAA HTTP/1.1" 200 103 "-" "Garmin Venu 2/8.11"
192.168.X.X - - [05/May/2022:09:01:51 -0400] "GET /server/json.server.php?auth=YYYY&timestamp=1651755710&action=handshake&user=USER&version=440000 HTTP/1.1" 200 512 "-" "Garmin Venu 2/8.11"
192.168.X.X - - [05/May/2022:09:01:51 -0400] "GET /server/json.server.php?date=1651160598&id=13&client=SubMusic-v0.1.24-yankee&action=record_play&auth=BBBB HTTP/1.1" 200 103 "-" "Garmin Venu 2/8.11"
192.168.X.X - - [05/May/2022:09:01:52 -0400] "GET /server/json.server.php?auth=ZZZZ&timestamp=1651755711&action=handshake&user=USER&version=440000 HTTP/1.1" 200 512 "-" "Garmin Venu 2/8.11"
```

**Comment by lemieuxl** [comment]
I see Ampache 5.3.2 was released 6 days ago. Do you want me to try it, or wait to gather more information for this issue?

**Comment by memen45** [comment]
You could try to copy the request in the browser and see what the response is from the server. Use the record_play request url after the last handshake request to make sure the auth parameter is valid. 

> I see Ampache 5.3.2 was released 6 days ago. Do you want me to try it, or wait to gather more information for this issue?

The issue was fixed 8 minutes after I created it yesterday, haha, so it will probably be fixed in the following 5.3.3 version!

**Comment by memen45** [comment]
No problem, great to hear sync is working again for you and thanks again for reporting!  Closing this issue for now.
